packages/eslint-plugin-office-addins/src/utils/utils.ts

```diff
@@ -72,12 +78,23 @@ function findOfficeApiReferencesInScope(scope: Scope): void {
   scope.childScopes.forEach(findOfficeApiReferencesInScope);
 }
 
+export function isCallingMethod(node: TSESTree.MemberExpression): boolean {
```

Comment:
I would name it 'isMethod' or 'isMethodCall'

Reply:
I like it `isCallingMethod` especially now after I put the `topLevelExpression` call inside the function, given that the argument of the functions is the object.
For example, in `range.getCell(0,0)`, the function receives `range` and it is not a method, it is just calling a method.

Reply:
Is the the method here is determining if the argument is a method call (i.e. is 'getCell' a method => true while is 'range' a method => false) or is it determining if the argument is the caller of the method (i.e. is 'range' calling a method => true while is 'getCell' calling a method => false).  The name should match what the returning T/F is indicating . . . and the logic can be inverted to match the name.

I'm thinking we are passing in the reference and finding out if it is a method call or not.

Reply:
Your passing in the argument 'getCell' (which is why you are trying to walk up the tree to find range) so the argument isn't the caller, it's the method itself and should return false, but you are finding the parent 'range' node, identifying it as the caller and returning true . . . so the argument doesn't match the name.

Reply:
I'm not passing `getCell`, the argument is `range`. I figured that the call to `findTopLevelExpression` is not needed and that is why you are probably getting confused with.
I updated the code to not have it.

Reply:
OK . . . I had to go back and spend sometime remembering the code and tree.  I was getting my tree backwards thinking it was walking backwards from the getCell call, but actually the first segments in a series of properties is the inner most node in the tree rather then other outermost (which is actually where the method call name ends up).  I think you are correct that the 'range' node is the node being tested, but that is also why you **do** need the findTopLevelExpression call.

In reviewing the code I see that "isLoadFunction" and "isLoadReference" are more along the line of what I'm think this should look like (in name and type usage) . . . if fact, I'm thinking that the difference between "isLoadReference" & "isLoadFunction" and what I think should be "isMethodReference" & "isMethodFunction" (or "isMethodCallReference") is really just the test for the identifier node "name" property.  I would think about following the established pattern with some reuse between the methods.

Reply:
Sure, I changed the designs to be consistent